benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/scheduler/TestTaskSchedulerIT.java

```diff
@@ -175,6 +175,58 @@ public void runRandomBreakdownExplorationUsers() throws Exception {
 
   }
 
+  @Test
+  public void runOneAtATimeExplorationUsersMemoryEnvironment() throws Exception {
+
+    String testName = "WfMSTest";
+    int expectedNumExperiments = 16;
+
+    String testID = testModelDAO.addTestModel(testName, user);
+
+    String testDefinitionString =
+        IOUtils.toString(TestFiles.getTestExplorationOneAtATimeUsersMemoryEnvironmentInputStream(),
+            StandardCharsets.UTF_8);
+
+    CountDownLatch countDownLatch =
+        setUpExplorationMocks(testID, expectedNumExperiments, testDefinitionString);
+
+    // handle in scheduler
+    testTaskScheduler.handleTestState(testID);
+
+    // wait long enough for all experiments to complete
+    countDownLatch.await(10, TimeUnit.SECONDS);
+
+    // wait for last task to finish
+    executorService.awaitTermination(1, TimeUnit.SECONDS);
+
+    // assert that all experiments have been executed
+    Assert.assertEquals(0, countDownLatch.getCount());
+
+    // assert that test has been set as TERMINATED
+    Assert.assertEquals(BenchFlowTestModel.BenchFlowTestState.TERMINATED,
+        testModelDAO.getTestState(testID));
+
+    // assert that the complete exploration space has been execution
+    List<Integer> expectedIndices = new ArrayList<>();
+    for (int i = 0; i < expectedNumExperiments; i++) {
+      expectedIndices.add(i);
+    }
+
```

Comment:
You should have assertion for the entire space, or better point me to the unit test where you do that.